lib/auth.js

```diff
@@ -99,12 +102,19 @@ export default class Auth {
      * @param {string} authObject.client_secret Client Secret from SFMC config
      * @param {number} authObject.account_id MID of Business Unit used for API Calls
      * @param {string} authObject.auth_url Auth URL from SFMC config
+     * @param {string} authObject.grant_type Grant Type. If present then access_token should be passed to the Auth object.
      * @param {string[]} [authObject.scope] Array of scopes used for requests
      * @param {object} options options for the SDK as a whole, for example collection of handler functions, or retry settings
      */
-    constructor(authObject, options) {
+    constructor(authObject, options, grant_type = GRANT_TYPE_CLIENT_CREDENTIALS) {
```

Comment:
Hey @fazbyte in docs above you say grant type is on authObject, but you pass as a param?  I think the authObject object is correct for this value to be honest. you can always set the default in the first line in case missing or use object.assign to set defaults and overwrite with provided values in authObject

Reply:
@DougMidgley , grant_type is not part of authObject. When the user logs in with the custom app, we then call the v2/token endpoint and after we get the response, we pass the response.data to the sfmc-sdk without changing it and pass the additional parameter grant_type='authorization_code'. This way it works with what you have already created and within the custom app where the user logs in. Let me know if this makes sense or what other change could be made.

Reply:
@DougMidgley I have completely removed the grant_type parameter now. However if authObject doesn't contain access_token, then it will kickoff the validations for client_id/client_secret. Let me know your feedback

Reply:
thank you @fazbyte - knowing Doug's schedule, he will review this next week. Right, @DougMidgley ? ;-)